### PR TITLE
[Release-2.x] Update publish github workflow to remove useless if check (#2102)

### PR DESCRIPTION
**Pull request by nainkunal933**
**Issue #:**

**Description of changes:**
Duplicate of https://github.com/aws/amazon-chime-sdk-js/pull/2102
**Testing:**

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
NA

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
NA

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
NA

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -96,13 +96,5 @@ jobs:
         run: |
           demo_current_version=$(.github/script/get-demo-browser-current-version)
           echo "Current demo version: " $demo_current_version
-          echo "Current released npm version:" ${{ steps.npm_version.outputs.npm_version }}
-          if [[ *$demo_current_version* != "^#${{ steps.npm_version.outputs.npm_version }}" ]];
-          then
-            echo "Demo npm version is not the same as the current released version"
-            exit 1
-          else
-            npm run build --app=meetingReadinessChecker
```
Comment (michhyun1) [inline .github/workflows/publish.yml:105]: we dont need this?

**Review by xuesichao — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (nainkunal933) [reply .github/workflows/publish.yml:105]: No. After looking at the workflow it turns out that this if statement will never be true.

Comment (nainkunal933) [reply .github/workflows/publish.yml:105]: This if statement's string comparison was the reason why the script was failing.

Comment (xuesichao) [reply .github/workflows/publish.yml:105]: Why we need to remove this:
>npm run build --app=meetingReadinessChecker

Comment (ltrung) [reply .github/workflows/publish.yml:105]: The deploy script already build it

**Pull request merged**